### PR TITLE
Open JTalk 1.08 ( new formula )

### PR DESCRIPTION
**Pull request by yawara**
I submitted OpenJTalk formula again.
OpenJTalk is a Japanese text-to-speech system.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
@yawara For future reference it's better to just push your fixes to the same branch so we can keep the conversation in one place.

File: Library/Formula/open-jtalk.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+class OpenJtalk < Formula
+  homepage "http://open-jtalk.sourceforge.net/"
+  url "https://downloads.sourceforge.net/project/open-jtalk/Open%20JTalk/open_jtalk-1.08/open_jtalk-1.08.tar.gz"
+  sha1 "34749abce5f8263ebbe9843b92407f4f0a742c66"
+
+  resource "hts_engine API" do
+    url "https://downloads.sourceforge.net/project/hts-engine/hts_engine%20API/hts_engine_API-1.09/hts_engine_API-1.09.tar.gz"
+    sha1 "1c264c2bd29c87f49ace2c5d2b2fcd6b5b44b12c"
+  end
+  resource "voice" do
```
Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: Add newline above this line.

File: Library/Formula/open-jtalk.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "-x", "#{prefix}/dic",
+      "-m", "#{prefix}/voice/mei/mei_normal.htsvoice",
+      "-ow", "out.wav",
+      "sample.txt"
+  end
+
+  def caveats; <<-EOS.undent
+    open_jtalk -x $(brew --prefix open-jtalk)/dic -m $(brew --prefix open-jtalk)/voice/mei/mei_normal.htsvoice -ow out.wav
+    EOS
+  end
```
Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: Can probably remove these `caveats`.

Comment (yawara) [inline Library/Formula/open-jtalk.rb]: I removed it.

File: Library/Formula/open-jtalk.rb (hunk 1 above)
Comment (yawara) [inline Library/Formula/open-jtalk.rb]: I inserted newline above line 10.

File: Library/Formula/open-jtalk.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    args = %W[
+      --with-hts-engine-header-path=#{include}
+      --with-hts-engine-library-path=#{lib}
+      --with-charset=UTF-8
+      --prefix=#{prefix}
+    ]
+
+    system "./configure", *args
+
+    inreplace "config.status", "-finput-charset=UTF-8 -fexec-charset=UTF-8", ""
```
Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/open-jtalk.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    system "make", "install"
+
+    resource("voice").stage do
+      mkdir_p "#{prefix}/voice/m100"
+      cp_r Dir["*"], "#{prefix}/voice/m100"
+    end
+    resource("mei").stage do
```
Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: newline above here.

File: Library/Formula/open-jtalk.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    system "./configure", *args
+
+    inreplace "config.status", "-finput-charset=UTF-8 -fexec-charset=UTF-8", ""
+
+    system "make", "install"
+
+    resource("voice").stage do
+      mkdir_p "#{prefix}/voice/m100"
+      cp_r Dir["*"], "#{prefix}/voice/m100"
```
Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: `"(prefix/"voice/m100").install Dir["*"]` will do both the above lines.

File: Library/Formula/open-jtalk.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    end
+    resource("mei").stage do
+      cp_r "Voice/mei", "#{prefix}/voice"
```
Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: `"(prefix/"voice").install "Voice/mei"`

File: Library/Formula/open-jtalk.rb (hunk 3 above)
Comment (yawara) [inline Library/Formula/open-jtalk.rb]: to Homebrew's upstream developers? if so, can i submitted through GitHub?

Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: To the `open-jtalk` developers, sorry.

Comment (yawara) [inline Library/Formula/open-jtalk.rb]: I see.

Comment (yawara) [inline Library/Formula/open-jtalk.rb]: @mikemcquaid gcc supports -fexec-charset and -finput-charset options but clang doesn't support them.
My patch eliminates these options from Makefiles.

[GCC compatibility: Ignore -finput_charset=UTF-8 argument. It is the default in Clang. Currently, we fail with an error.](http://reviews.llvm.org/D4347)
[ GCC compatibility: Ignore -fexec-charset=UTF-8 argument. It is the default in Clang. Reject other values. ](http://llvm.org/klaus/clang/commit/21c38cfcdf9ecc8aa5325bcdc46cc00f5a8e3015/)

Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: @yawara Please report that to the `open-jtalk` developers and add a link to your report in this formula file. Tahnks!

Comment (yawara) [inline Library/Formula/open-jtalk.rb]: @mikemcquaid I added the link to the archive of mails between me and the main maintainer of Open JTalk.
I am not sure if it is enough...

Comment (mikemcquaid) [inline Library/Formula/open-jtalk.rb]: That's great, thanks!

**Comment by DomT4** [comment]
If you're still interested here please fixup any comments highlighted by `brew audit --strict` and comment here. Thanks!

**Comment by yawara** [comment]
I executed `brew audit open-jtalk --strict` then got the follows.

```
==> brew style open-jtalk

1 file inspected, no offenses detected
```

**Comment by DomT4** [comment]
Is your Homebrew up-to-date? It should have told you to add a `desc` to the formula (80 characters or less).

**Comment by DomT4** [comment]
Ping.

**Comment by DomT4** [comment]
Presuming dead from lack of response. Please feel free to fix-up various comments and resubmit the PR if you're still interested, but closing here for now. Thanks for the submission!

**Pull request closed without merging**